libmuscle/python/libmuscle/manager/manager.py

```diff
@@ -121,6 +125,9 @@ def stop(self) -> None:
         """Shuts down the manager."""
         if self._instance_manager:
             self._instance_manager.shutdown()
+        self._deadlock_detector.shutdown()
+        # Note: don't join() deadlock detector, as this method may be called from the
+        # DeadlockDetector thread. join() would (ironically) deadlock the shutdown :)
```

Comment:
Now if only this were Java, then we could have a `class DeadlockDetectorShutdownDeadlockDetector` and it would be some kind of design pattern that we could put onto our CVs!

Reply:
😅 

I've actually found later in the Python docs that calling `Thread.join()` on your own thread raises a RuntimeError instead of deadlocking, so I doubly prefer that this is Python instead of Java 😜

Reply:
Well, there go our careers...